scripts/utilities/pyrescal.py

```diff
@@ -58,13 +155,83 @@ def run_simulation(self):
         os.close(w)
         
         # start a rescal process that can send data back
-        rescal = subprocess.Popen([self.rescal_executable] + [self.rescal_args], pass_fds=([w]))
+        rescal = subprocess.Popen(self.rescal_args, pass_fds=([w]))
```

Comment:
Since you're not polling for data, `subprocess.check_output()` would be more appropriate. It automagically gets the data and raises an exception if the callee closes unexpectedly.